hass_nabucasa/acme.py

```diff
@@ -409,7 +409,8 @@ async def issue_certificate(self) -> None:
             self._start_challenge, order
         )
 
-        for challenge in dns_challenges:
+        dns_entries = len(dns_challenges)
+        for idx, challenge in enumerate(dns_challenges):
```

Comment:
let's loop inside the `try`, so the final make sure it clean it up. Now if there is an exception on the first round, it never clean it up

Reply:
Ill move it out and do `dns_challenges[-1]` for the cleanup instead of all this 👍